sh_scrapy/extension.py

```diff
@@ -48,13 +49,24 @@ def spider_closed(self, spider, reason):
         self.pipe_writer.set_outcome(reason)
 
 
+_DOWNLOADER_MIDDLEWARE_WARNING = """\
+sh_scrapy.extension.HubstorageMiddleware functionality is now split between two new middlewares:
+
+- sh_scrapy.middlewares.HubstorageDownloaderMiddleware
+- sh_scrapy.middlewares.HubstorageSpiderMiddleware
+
+Please migrate to new middlewares.
+"""
+
+
 class HubstorageMiddleware(object):
 
     def __init__(self):
+        warnings.warn(_DOWNLOADER_MIDDLEWARE_WARNING)
```

Comment:
@dangra I doubt anyone did that because it makes little sense as this middleware is only used for communication with the platform - why would someone override that? I wasn't able to find sings of such usage anywhere. however I don't exclude this possibility, that's why I decided not to remove this class but add a deprecation warning - it's more descriptive that plain import error. if user notices any weird behavior caused by duplicated middlewares and errors / warnings he will have a useful hint in the logs.

based on assumption that it's unlikely to have such use cases I wouldn't bother too much with backwards compatibility between old middlewares and 2 new middlewares. it's tricky to make such changes backwards compatible if possible at all.

I'll update the PR to use the deprecation util, thanks